File: checkout/src/js/http/CourseClassesApi.ts
import {HttpService} from "../common/services/HttpService";
import {CommonError} from "../model/common/CommonError";
import {Course} from "../model/web/Course";
import {CourseClass} from "../model/web/CourseClass";
import {CourseClassesParams} from "../model/web/CourseClassesParams";
import {CoursesParams} from "../model/web/CoursesParams";

export class CourseClassesApi {
  constructor(private http: HttpService) {
  }

  getCourseClasses(courseClassesParams: CourseClassesParams): Promise<CourseClass[]> {
    return this.http.POST(`/v1/classes`, courseClassesParams);
  }
  getAvailableClasses(courseId: string): Promise<CourseClass[]> {
    return this.http.POST(`/v1/course/classes${courseId}`);
  }
  getCourses(coursesParams: CoursesParams): Promise<Course[]> {
    return this.http.POST(`/v1/courses`, coursesParams);
  }
}

File: checkout-api/src/main/groovy/ish/oncourse/willow/service/CourseClassesApi.java
package ish.oncourse.willow.service;

import ish.oncourse.willow.model.common.CommonError;
import ish.oncourse.willow.model.web.Course;
import ish.oncourse.willow.model.web.CourseClass;
import ish.oncourse.willow.model.web.CourseClassesParams;
import ish.oncourse.willow.model.web.CoursesParams;

import java.io.InputStream;
import java.io.OutputStream;
import java.util.List;
import java.util.Map;
import javax.ws.rs.*;
import javax.ws.rs.core.Response;
import javax.ws.rs.core.MediaType;

@Path("/")
public interface CourseClassesApi  {

    @POST
    @Path("/v1/course/classes/{courseId}")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @CollegeInfo
    List<CourseClass> getAvailableClasses(@PathParam("courseId") String courseId);

    @POST
    @Path("/v1/classes")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @CollegeInfo
    List<CourseClass> getCourseClasses(CourseClassesParams courseClassesParams);

    @POST
    @Path("/v1/courses")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @CollegeInfo
    List<Course> getCourses(CoursesParams coursesParams);
}

